static/js/components/AssignmentList.jsx

```diff
@@ -16,6 +16,22 @@ const AssignmentList = ({ assignments }) => {
   const { observingRunList } = useSelector((state) => state.observingRuns);
   const { instrumentList } = useSelector((state) => state.instruments);
 
+  // fetch all the requester ids before rendering the component
+  const requester_ids = assignments.map(
+    (assignment) => assignment.requester_id
+  );
+  const unique_requester_ids = [...new Set(requester_ids)];
+  unique_requester_ids.sort((a, b) => a - b);
+
+  // use useEffect to only send 1 fetchUser per User
+  useEffect(() => {
+    unique_requester_ids.forEach((id) => {
+      if (!users[id]) {
+        dispatch(UserActions.fetchUser(id));
+      }
+    });
+  }, [...unique_requester_ids, dispatch]);
```

Comment:
Any reason you're spreading unique requester ids in the dependencies array? You just need to specify the names of the variables that the hook uses. Also, looks like `users` should be added here.